src/azure-cli/azure/cli/command_modules/backup/commands.py

```diff
@@ -64,12 +66,20 @@ def load_command_table(self, _):
         g.custom_command('backup-now', 'backup_now', client_factory=backups_cf)
         g.custom_command('disable', 'disable_protection', confirmation=True)
         g.custom_command('enable-for-azurefileshare', 'enable_for_azurefileshare')
+        g.custom_command('enable-for-azurewl', 'enable_protection_for_azure_wl')
```

Comment:
I really recommend you to consider making azure files share, vm, azure work load and etc be a target parameter for enable command. With more and more resources you support, such kind of commands will increase greatly, which doesn't obey CLI convention.
Hope you can have a discussion internally and consider my suggestion, which will also be helpful for your future development.

Reply:
@Juliehzl : This is something that we might have to discuss at a strategic level. We have the proposed pattern in PS and that is not working for us. When we list the single command, ALL parameters are optional because they are required for some workload type and not required for another. When we onboarded to the CLI framework, this was discussed and agreed that a separate command for each workload is much better since it easily gives customer what is required for a particular workload type. Similar is the case for restore. Hope this explains the context.